src/ucp/api/ucp.h

```diff
@@ -4128,6 +4128,131 @@ typedef struct ucp_ep_attr {
+
+    /**
+     * Unpacked memory key for a remote memory endpoint.
+     */
+    ucp_rkey_h rkey;
+} ucp_mem_list_elem_t;
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Descriptor list create parameters field mask.
+ *
+ * The enumeration allows specifying which fields in @ref ucp_mem_list_create_params_t
+ * are presents. It is used to enable backward compatibility support.
+ */
+enum ucp_mem_list_create_params_field {
+    UCP_MEM_LIST_CREATE_PARAMS_FIELD_ELEMENTS     = UCS_BIT(0), /**< Elements array base address */
+    UCP_MEM_LIST_CREATE_PARAMS_FIELD_ELEMENT_SIZE = UCS_BIT(1), /**< Element size in bytes */
+    UCP_MEM_LIST_CREATE_PARAMS_FIELD_NUM_ELEMENTS = UCS_BIT(2)  /**< Number of elements */
+};
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Descriptor list create parameters.
+ *
+ * The structure defines the parameters that can be used to create a handle
+ * with @ref ucp_gpu_mem_list_create.
+ */
+typedef struct ucp_mem_list_create_params {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_mem_list_create_params_field.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t                  field_mask;
+
+    /**
+     * Size in bytes of one descriptor element, for backward compatibility.
+     */
+    size_t                    element_size;
+
+    /**
+     * Number of elements presents in @ref ucp_mem_list_elem_t.
+     */
+    size_t                    num_elements;
+
+    /**
+     * Base address for the array of descriptor elements.
+     */
+    const ucp_mem_list_elem_t *elements;
+} ucp_mem_list_create_params_t;
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Descriptor list create function for batched RMA operations.
+ *
+ * This function creates and populates a descriptor list handle using parameters
+ * inputs from @ref ucp_mem_list_create_params_t. This descriptor is created for
+ * a given remote endpoint. It can be used on a GPU using the corresponding
+ * device functions.
+ *
+ * It can be used repeatedly, until finally released by calling @ref
+ * ucp_gpu_mem_list_release.
+ *
+ * @param [in]  ep        Remote endpoint handle
+ * @param [in]  params    Parameters used to create the created handle
+ * @param [out] handle    Exported descriptor list handle
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_gpu_mem_list_create(ucp_ep_h ep,
+                                     const ucp_mem_list_create_params_t *params,
+                                     ucp_gpu_mem_list_handle_h *handle);
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Release function for a descriptor list handle.
+ *
+ * This function releases the handle that was created using @ref
+ * ucp_gpu_mem_list_create.
+ *
+ * @param [in] handle     Exported handle to release
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_gpu_mem_list_release(ucp_gpu_mem_list_handle_h handle);
```

Comment:
```suggestion
void ucp_gpu_mem_list_release(ucp_gpu_mem_list_handle_h handle);
```
Please replace by `void`. This method should return `void`, just like other similar UCP API methods.